server/src/model/document.ts

```diff
@@ -37,47 +36,83 @@ export class Document {
   id: number;
   title: string;
   description: string;
-  coordinates: Coordinates;
-  scale: Scale;
   type: DocumentType;
+  scale: Scale;
+  stakeholders?: Stakeholder[];
+  coordinates?: Coordinates;
+  issuanceDate?: Dayjs;
 
   constructor(
     id: number,
     title: string,
     description: string,
-    coordinates: Coordinates,
-    scale: Scale,
     type: DocumentType,
+    scale: Scale,
+    stakeholders: Stakeholder[] | undefined = undefined,
+    coordinates: Coordinates | undefined = undefined,
+    issuanceDate: Dayjs | undefined = undefined,
   ) {
     this.id = id;
     this.title = title;
     this.description = description;
-    this.coordinates = coordinates;
-    this.scale = scale;
     this.type = type;
+    this.scale = scale;
+    this.stakeholders = stakeholders;
+    this.coordinates = coordinates;
+    this.issuanceDate = issuanceDate;
   }
 
   private static fromDatabaseRow(dbRow: DocumentDbRow): Document {
-    const { id, title, description, coordinates, scale, type } = dbRow;
-    assert(typeof id === "number");
+    const {
+      id,
+      title,
+      description,
+      type,
+      scale_type,
+      scale_ratio,
+      stakeholders,
+      coordinates,
+      issuanceDate,
+    } = dbRow;
     assert(typeof title === "string");
     assert(typeof description === "string");
-    assert(typeof scale === "string");
     assert(typeof type === "string");
-    // TODO: array asserts are missing
+    assert(typeof scale_type === "string");
+    assert(!issuanceDate || issuanceDate instanceof Date);
 
-    return new Document(id, title, description, coordinates, scale, type);
+    const scale: Scale = Scale.fromDatabaseRow({
+      scale_type,
+      scale_ratio,
+    } as ScaleRow);
+
```

Comment:
Don't know if the separation of document and scale is really helpful. Scale doesn't have a separate row in the database